src/plugins/search-disabled-repos.py

```diff
@@ -0,0 +1,110 @@
+# Copyright (C) 2015  Red Hat, Inc.
+#
+# Authors: Valentina Mukhamedzhanova <vmukhame@redhat.com>
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# the GNU General Public License v.2, or (at your option) any later version.
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY expressed or implied, including the implied warranties of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.  You should have received a copy of the
+# GNU General Public License along with this program; if not, write to the
+# Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+# source code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission of
+# Red Hat, Inc.
+
+import logging
+import os
+
+from yum.plugins import TYPE_CORE, TYPE_INTERACTIVE
+from yum.constants import TS_INSTALL_STATES
+
+requires_api_version = '2.7'
+plugin_type = (TYPE_CORE, TYPE_INTERACTIVE)
+
+PLUGIN_CONF_PATH = '/etc/yum/pluginconf.d/search-disabled-repos.conf'
+
+attempted = False
+old_enabled_repos = []
+
+
+def postresolve_hook(conduit):
+    if not conduit.missing_requires:
+        return
+
+    repo_storage = conduit.getRepos()
+    disabled_repos = set((repo for repo in repo_storage.repos.values() if not repo.enabled and is_repo_important(repo)))
+    if not disabled_repos:
+        return
+
+    notify_only = conduit.confBool('main', 'notify_only', default=True)
+    if notify_only:
+        suggest_enabling(conduit)
+        return
+
+    if prompt_temporarily_enable_repos(conduit):
+        conduit.info(logging.DEBUG, 'Depsolving failed due to missing dependencies, temporarily enabling repos...')
+        global attempted
+        attempted = True
+        global old_enabled_repos
+        old_enabled_repos = set((repo.id for repo in repo_storage.listEnabled()))
```

Comment:
In your example, rhel-7-server-rpms will get enabled by the plugin.

The idea behind this RFE is to make it easy for customers to resolve missing dependency problems. I believe the current implementation is consistent with this idea.

If we make not_important_suffixes configurable, it could be used to exclude 'rhel-7-server-rpms' from the list of repos to enable.